### PR TITLE
Panic kick

### PR DESCRIPTION
**Pull request by JNeiger**
Pivot kick kicks early when robot is close

### CODE REVIEW

File: soccer/gameplay/skills/pivot_kick.py [hunk 1]
```diff
@@ -124,6 +129,55 @@ def facing_opp_goal(self):
...
+        if (self.robot is None):
+            return False
+
+        # Closest opp robot in any direction
+        # To us, not the ball
+        # May want to change        
+        closest_opp_robot = None
+        closest_opp_dist = float("inf")
+        for r in main.their_robots():
+            if ((r.pos - self.robot.pos).mag() < closest_opp_dist):
```
Comment (tjones320) [inline soccer/gameplay/skills/pivot_kick.py]: Do you want to use robot position or ball position

File: soccer/gameplay/skills/pivot_kick.py [hunk 2]
```diff
@@ -124,6 +129,55 @@ def facing_opp_goal(self):
...
+        facing_their_side = robocup.Point.direction(self.robot.angle).y > 0# self.robot.angle > 0
+
+        ret = (facing_their_side and
+               robot_in_range and
+               robot_in_front and
+               does_hit_robot and
+               doesnt_hit_mouth)
+
+        if ret:
+            print("Panic kick")
```
Comment (tjones320) [inline soccer/gameplay/skills/pivot_kick.py]: Do you want to keep this print statement?

File: soccer/gameplay/skills/pivot_kick.py [hunk 3]
```diff
@@ -124,6 +129,55 @@ def facing_opp_goal(self):
...
+        # May want to change        
+        closest_opp_robot = None
+        closest_opp_dist = float("inf")
+        for r in main.their_robots():
+            if ((r.pos - self.robot.pos).mag() < closest_opp_dist):
+                closest_opp_robot = r
+                closest_opp_dist = (r.pos - self.robot.pos).mag()
+
+        # Only do this if a robot is in range
+        robot_in_range = closest_opp_dist < 5
```
Comment (tjones320) [inline soccer/gameplay/skills/pivot_kick.py]: Why five meters?

File: soccer/gameplay/skills/pivot_kick.py [hunk 4]
```diff
@@ -124,6 +129,55 @@ def facing_opp_goal(self):
...
+        robot_dir = (closest_opp_robot.pos - self.robot.pos)
+
+        # Only trigger if they are infront of us
+        robot_in_front = aim_dir.dot(robot_dir) > 0
+
+        closest_pt = robocup.Line(self.robot.pos,
+                                  self.robot.pos + aim_dir).nearest_point(
+                                      closest_opp_robot.pos)
+
+        does_hit_robot = (closest_opp_robot.pos - closest_pt).mag() < constants.Robot.Radius
```
Comment (tjones320) [inline soccer/gameplay/skills/pivot_kick.py]: so are we only kicking if they're one robot distance away?

File: soccer/gameplay/skills/pivot_kick.py (hunk 3 above)
Comment (JNeiger) [reply soccer/gameplay/skills/pivot_kick.py]: Should be .2

**Pull request merged**